packages/SwingSet/test/test-kernel.js

```diff
@@ -32,7 +32,7 @@ function emptySetup(_syscall) {
 }
 
 function makeEndowments() {
-  return { setImmediate, hostStorage: makeMemorySwingStore().storage };
+  return { setImmediate, hostStorage: makeSwingStore(null, true).storage };
```

Comment:
These `(null, true)` feel a little magic to me, not something one can guess the meaning of from the call site. From context I'm guessing `true` means to erase the contents (if any) and maybe tolerate a missing file. Neither seems super relevant for an in-memory store.

Would it help anything if these packages exported two separate functions: initialize and open, both of which take a directory (and no other arguments), and the filesystem form of the latter would refuse to run unless the former had been run first?

LMDB on disk:

```js
import { initialize, open } from '@agoric/swing-store-lmdb';
initialize(dirname); // erases if present, creates in all cases
const db = open(dirname); // insists already present
```

json-lines on disk:

```js
import { open } from '@agoric/swing-store-simple';
initialize(dirname); // erases if present, creates in all cases
const db = open(dirname); // insists already present
```

non-persistent Map in memory:

```js
import { open } from '@agoric/swing-store-simple';
const db = open(); // non-persistent
```

?

Reply:
Yah, it looked a little weird to me too.  One thing that isn't apparent either is that the flag is actually irrelevant to the in-memory case (since you're always going to get a new one no matter what).  I used `true` because it is being initialized after all.  But you could just write `makeSwingStore(null)`, which would dispense with the flag when it's meaningless, at the cost of possibly making the API a bit more mysterious.

I'm not sure where I come down on the initialize/open alternative.  On the one hand, it feels clunkier, since the two functions are largely useless independently (the exception being the simple store open() with no parameters), and having to repeat the dir name seems like a potential foot gun.  On the other hand it _is_ more explicit about what's going on, which has significant value that I like a lot.  Lemme sleep on this one and we'll decide it in the morning.

Reply:
Thinking about this a bit more, I think you could fold `open` into `initialize`:
```
const db = initialize(dirname); // erases if present, creates if not, returns db object
```
or
```
const db = initialize(); // for the in-memory case`
```
vs:
```
const db = open(dirname); // only works if the db is already there
```

In this factoring, it might be better to name the first one `create` (or `creat` if we want to make an ironic homage to Ken Thompson, but that would be wrong).